jobs/resource_job_test.go

```diff
@@ -429,6 +430,161 @@ func TestResourceJobCreateNWorkers(t *testing.T) {
 	assert.Equal(t, "789", d.Id())
 }
 
+func TestResourceJobCreateFromGitSource(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "POST",
+				Resource: "/api/2.1/jobs/create",
+				ExpectedRequest: JobSettings{
+					ExistingClusterID: "abc",
+					Tasks: []JobTaskSettings{
+						{
+							TaskKey: "b",
+							NotebookTask: &NotebookTask{
+								NotebookPath: "/GitSourcedNotebook",
+							},
+						},
+					},
+					Name:              "GitSourceJob",
+					MaxConcurrentRuns: 1,
+					GitSource: &GitSource{
+						Url:      "https://github.com/databrickslabs/terraform-provider-databricks",
+						Tag:      "0.4.8",
+						Provider: "gitHub",
+					},
+				},
+				Response: Job{
+					JobID: 789,
+				},
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/jobs/get?job_id=789",
+				Response: Job{
+					JobID: 789,
+					Settings: &JobSettings{
+						// good enough for mock
+						ExistingClusterID: "abc",
+						Tasks: []JobTaskSettings{
+							{
+								TaskKey: "b",
+							},
+						},
+						Name:              "GitSourceJob",
+						MaxConcurrentRuns: 1,
+					},
+				},
+			},
+		},
+		Create:   true,
+		Resource: ResourceJob(),
+		HCL: `existing_cluster_id = "abc"
+		max_concurrent_runs = 1
+		name = "GitSourceJob"
+
+		git_source {
+			url = "https://github.com/databrickslabs/terraform-provider-databricks"
+			tag = "0.4.8"
+		}
+
+		task {
+			task_key = "b"
+
+			notebook_task {
+				notebook_path = "/GitSourcedNotebook"
+			}
+		}`,
+	}.Apply(t)
```

Comment:
use either `.ApplyNoError` or `.ApplyAndExpectData`. two lines in the end look a bit ugly :)